config/src/main/java/org/axonframework/config/Configurer.java

```diff
@@ -212,6 +212,45 @@ Configurer configureCorrelationDataProviders(
     <C> Configurer registerComponent(@Nonnull Class<C> componentType,
                                      @Nonnull Function<Configuration, ? extends C> componentBuilder);
 
+
+    /**
+     * Registers a {@link ComponentDecorator decorator} for a component type. For any component that is created by the
+     * configuration that matches the {@code componentType}, the {@code decorator} will be called. It's up to the
+     * decorator to decide to return the original component or a wrapped version of it.
+     *
+     * @param componentType                     The declared type of the component, typically an interface
+     * @param decoratorFunction                 The decorator function for this component
+     * @param registerOriginalLifeCycleHandlers Whether the original component's lifecycle handlers should be
+     *                                          registered.
+     * @param <C>                               The declared type of component
+     * @return the current instance of the Configurer, for chaining purposes
+     */
+    default <C> Configurer registerComponentDecorator(@Nonnull Class<C> componentType,
```

Comment:
I noticed the order of parameters here is different than for the constructor of the ComponentTypeSafeDecorator. The boolean is probably last here as it is an optional parameter?